### PR TITLE
feat: solidity verifier usage example

### PR DESCRIPTION
**Pull request by merklefruit**
This PR adds a simple example of using the library to prove the inclusion of a transaction in an Ethereum transaction trie. The example takes in a `txHash`, generates a proof by interacting with the CLI binary via FFI, parses it and feeds it to a `MerkleTrie` verifier after RLP-encoding it correctly.

Note: the test has to be run with `forge test --ffi` to enable FFI usage.

### CODE REVIEW

**Comment by rkdud007** [comment]
thanks for neat example, here is two comments i have: 

- currently default rpc is public node as this example tx hash potentially be deprecated via public node rpc. would be nice to accept rpc url as .env param on test 
- can we mention on readme : to run this example `forge test --ffi`

File: examples/eth/src/Prover.sol [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import {MerkleTrie} from "./lib/MerkleTrie.sol";
+
+contract Prover {
+    constructor() {}
+
+    function get(bytes memory _key, bytes memory _proof, bytes32 _root) public pure returns (bool, bytes memory) {
+        return MerkleTrie.get(_key, _proof, _root);
+    }
+
+    function verifyInclusionProof(bytes memory _key, bytes memory _value, bytes memory _proof, bytes32 _root)
```
Comment (rkdud007) [inline examples/eth/src/Prover.sol:13]: seems this function is not used, can we call this function on the example?

**Comment by merklefruit** [comment]
@rkdud007 added docs and a custom settable RPC endpoint which falls back to a good default.

**Comment by rkdud007** [comment]
lgtm, thanks for example @merklefruit :)

**Pull request merged**